# problem1

In [1]:
c1 <- c(3135,3001,2865,2890)
c2 <- c(3210,3311,2975,3150)
c3 <- c(2815,2910,2985,3050)
c4 <- c(2619,2706,2612,2765)
y <- c(c1,c2,c3,c4)
n <- rep(4,4)
group <- rep(1:4,n)
data = data.frame(y = y, group = factor(group))
fit = lm(y ~ group, data)
anova(fit)

Df Sum Sq   Mean Sq   F value  Pr(>F)      
group      3 480275.2 160091.73 12.63776 0.0005045012
Residuals 12 152012.7  12667.73       NA           NA

In [2]:
data

y    group
1  3135 1    
2  3001 1    
3  2865 1    
4  2890 1    
5  3210 2    
6  3311 2    
7  2975 2    
8  3150 2    
9  2815 3    
10 2910 3    
11 2985 3    
12 3050 3    
13 2619 4    
14 2706 4    
15 2612 4    
16 2765 4

In [128]:
(y-mean(y))%*%(y-mean(y))

[,1]    
[1,] 632287.9

In [129]:
u2 <- c(1,-1,0,0)
u3 <- c(1,1,-2,0)
u4 <- c(1,1,1,-3)
ybar <- c(mean(c1),mean(c2),mean(c3),mean(c4))
residMS <- 12667.73

In [130]:
(ybar%*%u2)^2*4/u2%*%u2

[,1]    
[1,] 71253.12

In [131]:
(ybar%*%u2)^2*4/u2%*%u2/residMS

[,1]    
[1,] 5.624775

In [132]:
pf(5.624775 ,1,12,lower=F)

[1] 0.03529459

In [133]:
(ybar%*%u3)^2*4/u3%*%u3

[,1]    
[1,] 43095.38

In [134]:
(ybar%*%u3)^2*4/u3%*%u3/residMS

[,1]    
[1,] 3.401981

In [135]:
pf(3.401981 ,1,12,lower=F)

[1] 0.0899335

In [136]:
(ybar%*%u4)^2*4/u4%*%u4

[,1]    
[1,] 365926.7

In [137]:
(ybar%*%u4)^2*4/u4%*%u4/residMS

[,1]    
[1,] 28.88652

In [138]:
pf(28.88652,1,12,lower=F)

[1] 0.0001668047

## (b) vector法ANOVA table

ANOVA table   | df    | Sum Sq | Mean Sq | F value|	Pr(>F)
--------------|:-----:|-------:|--------:|-------:|----------------
group         | 3 |  480275.2  |   160091.73| 12.63776   |  0.0005045012
c1:u1=u2      | 1 | 71253.12 |  71253.12    |5.624775|0.035294
c2:u1+u2=2u3  | 1 | 43095.38| 43095.38   |3.401981|0.0899334
c3:u1+u2+u3=3u4 | 1 |365926.7 | 365926.7  |28.88652|0.00016680
Residuals	  |12 |152012.7	   |	12667.73	 |NA	  |NA

## (C) Test the hypothesis that mixing techniques affect the strength of the cement. Use $\alpha$ =0.05. 

$H_{0}:\mu_{i}全等$  
$H_1:\mu_{i}不全等$

P value <0.05，推翻虛無假設，mixing techniques對Tensile strength的影響不全等。

## (d) If the mixing technique affect the strength of the cement. Please specify which mixing their differences.

$H_{0}:\mu_{1}=\mu_{2}$  
$H_1:\mu_{1}\ne\mu_{2}$  
Pvalue = 0.0352 ,推翻虛無假設,$\mu_{1}\ne\mu_{2}$  
***
$H_{0}:\mu_{2}=\mu_{3}$     
$H_1:\mu_{2}\ne\mu_{3}$   
Pvalue = 0.01655291 ,推翻虛無假設,$\mu_{2}\ne\mu_{3}$    

---
$H_{0}:\mu_{3}=\mu_{4}$    
$H_1:\mu_{3}\ne\mu_{4}$  
Pvalue = 0.006071094 ,推翻虛無假設,$\mu_{3}\ne\mu_{4}$  

---
$H_{0}:\mu_{1}=\mu_{4}$    
$H_1:\mu_{1}\ne\mu_{4}$  
Pvalue = 0.002847966 ,推翻虛無假設,$\mu_{1}\ne\mu_{4}$  

In [139]:
k2 <- c(0,1,-1,0)
k3 <- c(0,0,1,-1)
k4 <- c(1,0,0,-1)

In [140]:
pf((ybar%*%k2)^2*4/k2%*%k2/residMS,1,12,lower=F)

[,1]      
[1,] 0.01655291

In [141]:
pf((ybar%*%k3)^2*4/k3%*%k3/residMS,1,12,lower=F)

[,1]       
[1,] 0.006071094

In [142]:
pf((ybar%*%k4)^2*4/k4%*%k4/residMS,1,12,lower=F)

[,1]       
[1,] 0.002847966

# problem2

In [143]:
c1 <- c(491,579,451,521,503)
c2 <- c(588,502,550,520,470)
c3 <- c(533,628,501,537,561)
y <- c(c1,c2,c3)
n <- rep(5,3)
group <- rep(1:3,n)
data = data.frame(y = y, group = factor(group))
fit = lm(y ~ group, data)
anova(fit)

Df Sum Sq Mean Sq F value  Pr(>F)   
group      2  4690  2345    1.083141 0.3694471
Residuals 12 25980  2165          NA        NA

In [144]:
(y-mean(y))%*%(y-mean(y))

[,1] 
[1,] 30670

In [145]:
u2 <- c(1,-1,0)
u3 <- c(1,1,-2)
ybar <- c(mean(c1),mean(c2),mean(c3))

In [146]:
(ybar%*%u2)^2*5/u2%*%u2

[,1] 
[1,] 722.5

In [147]:
(ybar%*%u3)^2*5/u3%*%u3

[,1]  
[1,] 3967.5

In [148]:
pf(1.083141,2,12,lower=F)

[1] 0.369447

In [149]:
pf(722.5/2165,1,12,lower=F)

[1] 0.5741577

In [150]:
pf(3967.5/2165,1,12,lower=F)

[1] 0.2007729

## (b) vector法ANOVA table

ANOVA table   | df    | Sum Sq | Mean Sq | F value|	Pr(>F)
--------------|:-----:|-------:|--------:|-------:|----------------
group         | 2 |  4690    |    2345   | 1.083141   |  0.3694471
c1:u1=u2      | 1 |  722.5   |  722.5    |0.33371        |  0.574157
c2:u1+u2=2u3  | 1 | 3967.5   |  3967.5   | 1.83256       | 0.200772
Residuals	  |12 |	25980    |	2165	 |NA	  |NA


## (c)高低氮肥對產量的影響相同？

### vector=(1,1,1,1,1,-1,-1,-1,-1,-1,0,0,0,0,0)

$H_{0}:\mu_{高氮肥}=\mu_{低氮肥}$  
$H_1:\mu_{高氮肥}\ne\mu_{低氮肥}$

In [151]:
(ybar%*%u2)^2*5/u2%*%u2

[,1] 
[1,] 722.5

In [152]:
pf(722.5/2165,1,12,lower=F)

[1] 0.5741577

p value>0.05,無法推翻高低氮肥對產量的影響相同的虛無假設

## (d)氮肥與鉀肥對產量的影響相同？

### vector=(1,1,1,1,1,1,1,1,1,1,-2,-2,-2,-2,-2)

$H_{0}:\mu_{高氮肥}+\mu_{低氮肥}=\mu_{鉀肥}$  
$H_1:\mu_{高氮肥}+\mu_{低氮肥}\ne\mu_{鉀肥}$

In [153]:
(ybar%*%u3)^2*5/u3%*%u3

[,1]  
[1,] 3967.5

In [1]:
pf(3967.5/2165,1,12,lower=F)

[1] 0.2007729

p value>0.05,無法推翻氮肥和鉀肥對產量的影響相同的虛無假設

In [17]:
treat<-c("A","A","A","B","B","C","C","C","C")
Y<-c(3,5,4,8,5,3,2,5,3);
test<-data.frame(treat,Y);
prob2.aov<-aov(Y~treat);
summary(prob2.aov);
# Building the contrast matrix
c1<-c(0,2,-1); # vitamin B vs. C
c2<-c(-2,1,1); # vitamin A vs. B + C
mat<-cbind(c1,c2);
# Assign the matrix to the classification variable, treat.
contrasts(test$treat)<-mat
contrast.mod<-aov(Y~treat,test);
summary(contrast.mod,split=list(treat=list("b vs c"=1,"a vs b+c"=2)));
test.mod<-lm(Y~treat,test)
summary(test.mod);

            Df Sum Sq Mean Sq F value Pr(>F)  
treat        2  14.31   7.153   3.815 0.0853 .
Residuals    6  11.25   1.875                 
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

                  Df Sum Sq Mean Sq F value Pr(>F)  
treat              2 14.306   7.153   3.815 0.0853 .
  treat: b vs c    1 14.083  14.083   7.511 0.0337 *
  treat: a vs b+c  1  0.222   0.222   0.119 0.7424  
Residuals          6 11.250   1.875                 
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1


Call:
lm(formula = Y ~ treat, data = test)

Residuals:
   Min     1Q Median     3Q    Max 
 -1.50  -1.00  -0.25   1.00   1.75 

Coefficients:
            Estimate Std. Error t value Pr(>|t|)    
(Intercept)   4.2222     0.4564   9.250 9.02e-05 ***
treatc1       1.0833     0.3953   2.741   0.0337 *  
treatc2       0.1111     0.3227   0.344   0.7424    
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

Residual standard error: 1.369 on 6 degrees of freedom
Multiple R-squared:  0.5598,	Adjusted R-squared:  0.413 
F-statistic: 3.815 on 2 and 6 DF,  p-value: 0.08531


In [41]:
treat<-c("A","A","A","B","B","C","C","C","C")
Y<-c(3,5,4,8,5,3,2,5,3);
test<-data.frame(treat,Y);
prob2.aov<-aov(Y~treat);
summary(prob2.aov);
# Building the contrast matrix
c1<-c(0,2,-1); # vitamin B vs. C
c2<-c(4,0,-3); # vitamin A vs. C
mat<-cbind(c1,c2);
# Assign the matrix to the classification variable, treat.
contrasts(test$treat)<-mat
contrast.mod<-aov(Y~treat,test);
summary(contrast.mod,split=list(treat=list("b vs c"=1,"a vs c"=2)));
test.mod<-lm(Y~treat,test)
summary(test.mod);
#若向量不正交，除了第一個比較(treat: b vs c)之外接不能參考


            Df Sum Sq Mean Sq F value Pr(>F)  
treat        2  14.31   7.153   3.815 0.0853 .
Residuals    6  11.25   1.875                 
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

                Df Sum Sq Mean Sq F value Pr(>F)  
treat            2 14.306   7.153   3.815 0.0853 .
  treat: b vs c  1 14.083  14.083   7.511 0.0337 *
  treat: a vs c  1  0.222   0.222   0.119 0.7424  
Residuals        6 11.250   1.875                 
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1


Call:
lm(formula = Y ~ treat, data = test)

Residuals:
   Min     1Q Median     3Q    Max 
 -1.50  -1.00  -0.25   1.00   1.75 

Coefficients:
            Estimate Std. Error t value Pr(>|t|)    
(Intercept)  4.22222    0.45644   9.250 9.02e-05 ***
treatc1      1.13889    0.42696   2.667   0.0372 *  
treatc2     -0.05556    0.16137  -0.344   0.7424    
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

Residual standard error: 1.369 on 6 degrees of freedom
Multiple R-squared:  0.5598,	Adjusted R-squared:  0.413 
F-statistic: 3.815 on 2 and 6 DF,  p-value: 0.08531


In [20]:
contrasts(test$treat)

c1 c2
A  0  4
B  2  0
C -1 -3

In [22]:
contrast.mod<-aov(Y~treat,test);

In [23]:
contrast.mod

Call:
   aov(formula = Y ~ treat, data = test)

Terms:
                   treat Residuals
Sum of Squares  14.30556  11.25000
Deg. of Freedom        2         6

Residual standard error: 1.369306
Estimated effects may be unbalanced

In [59]:
Y <- c(39.2,40.4,45.3,46.3)
B <- c(1,2,1,2)
T <- c("c","c","f","f")
B <- factor(B)
T <- factor(T)
test<-data.frame(T,B,Y);

In [60]:
prob2.aov<-aov(Y~B+T);
summary(prob2.aov);

            Df Sum Sq Mean Sq F value Pr(>F)  
B            1   1.21    1.21     121 0.0577 .
T            1  36.00   36.00    3600 0.0106 *
Residuals    1   0.01    0.01                 
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

In [16]:
u2 <- c(0,1,-1)
ybar <- c(mean(c1),mean(c2),mean(c3))
(ybar%*%u2)^2*4/u2%*%u2

[,1]
[1,] NA

In [ ]:
day<-
  c("Day1","Day2","Day3","Day4");
sol<-rep(c(1:3),each=4);
weight<-c(13,22,18,39,
          16,24,17,44,
          5,4,1,22);
data<-data.frame(day,sol,weight);
data$day<-as.factor(data$day);
data$sol<-as.factor((data$sol));
data.mod<-aov(weight~day + sol,data=data);
summary(data.mod);

c1<-c(1,-1,0); # 
c2<-c(1,1,-2); #
mat<-cbind(c1,c2);
# Assign the matrix to the classification variable, treat.
contrasts(data$sol)<-mat
contrast.mod<-aov(weight~sol,data);
summary(contrast.mod,split=list(sol=list("sol 1 vs sol2"=1,"sol 1+2 vs 3"=2)));

c1<-c(0,1,-1); # 
c2<-c(-2,1,1); #
mat<-cbind(c1,c2);
# Assign the matrix to the classification variable, treat.
contrasts(data$sol)<-mat
contrast.mod<-aov(weight~sol,data);
summary(contrast.mod,split=list(sol=list("sol 2 vs sol 3"=1,"sol 2+3 vs 1"=2)));

c1<-c(1,0,-1); # 
c2<-c(1,-2,1); #
mat<-cbind(c1,c2);
# Assign the matrix to the classification variable, treat.
contrasts(data$sol)<-mat
contrast.mod<-aov(weight~sol,data);
summary(contrast.mod,split=list(sol=list("sol 1 vs sol 3"=1,"sol 1+3 vs 2"=2)));

In [26]:
c1<-c(1,-1,0,0); # 
c2<-c(1,1,-2,0);
c3<-c(1,1,1,-3)#
mat<-cbind(c1,c2,c3);
# Assign the matrix to the classification variable, treat.
contrasts(data$day)<-mat
contrast.mod<-aov(weight~day,data);
summary(contrast.mod,split=list(day=list("day 1 vs day2 "=1,"day 1+2 vs 3"=2,"day 1+2+3 vs 4"=3)));

c1<-c(0,1,-1,0); # 
c2<-c(1,1,-2,0);
c3<-c(1,1,1,-3)#
mat<-cbind(c1,c2,c3);
# Assign the matrix to the classification variable, treat.
contrasts(data$day)<-mat
contrast.mod<-aov(weight~day,data);
summary(contrast.mod,split=list(day=list("day 2 vs day3 "=1,"day 1+2 vs 3"=2,"day 1+2+3 vs 4"=3)));

c1<-c(0,0,1,-1); # 
c2<-c(1,1,-2,0);
c3<-c(1,1,1,-3)#
mat<-cbind(c1,c2,c3);
# Assign the matrix to the classification variable, treat.
contrasts(data$day)<-mat
contrast.mod<-aov(weight~day,data);
summary(contrast.mod,split=list(day=list("day 3 vs day4 "=1,"day 1+2 vs 3"=2,"day 1+2+3 vs 4"=3)));

c1<-c(1,0,0,-1); # 
c2<-c(1,1,-2,0);
c3<-c(1,1,1,-3)#
mat<-cbind(c1,c2,c3);
# Assign the matrix to the classification variable, treat.
contrasts(data$day)<-mat
contrast.mod<-aov(weight~day,data);
summary(contrast.mod,split=list(day=list("day 1 vs day4 "=1,"day 1+2 vs 3"=2,"day 1+2+3 vs 4"=3)));

c1<-c(0,-1,0,1); # 
c2<-c(1,1,-2,0);
c3<-c(1,1,1,-3)#
mat<-cbind(c1,c2,c3);
# Assign the matrix to the classification variable, treat.
contrasts(data$day)<-mat
contrast.mod<-aov(weight~day,data);
summary(contrast.mod,split=list(day=list("day 2 vs day4 "=1,"day 1+2 vs 3"=2,"day 1+2+3 vs 4"=3)));

c1<-c(1,0,-1,0); # 
c2<-c(1,1,-2,0);
c3<-c(1,1,1,-3)#
mat<-cbind(c1,c2,c3);
# Assign the matrix to the classification variable, treat.
contrasts(data$day)<-mat
contrast.mod<-aov(weight~day,data);
summary(contrast.mod,split=list(day=list("day 1 vs day3 "=1,"day 1+2 vs 3"=2,"day 1+2+3 vs 4"=3)));

                      Df Sum Sq Mean Sq F value Pr(>F)  
day                    3 1106.9   369.0   3.908 0.0547 .
  day: day 1 vs day2   1   42.7    42.7   0.452 0.5204  
  day: day 1+2 vs 3    1    8.0     8.0   0.085 0.7784  
  day: day 1+2+3 vs 4  1 1056.2  1056.2  11.187 0.0102 *
Residuals              8  755.3    94.4                 
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

                      Df Sum Sq Mean Sq F value Pr(>F)  
day                    3 1106.9   369.0   3.908 0.0547 .
  day: day 2 vs day3   1   32.7    32.7   0.346 0.5726  
  day: day 1+2 vs 3    1   18.0    18.0   0.191 0.6739  
  day: day 1+2+3 vs 4  1 1056.2  1056.2  11.187 0.0102 *
Residuals              8  755.3    94.4                 
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

                      Df Sum Sq Mean Sq F value Pr(>F)  
day                    2 1064.2   532.1   6.001 0.0221 *
  day: day 3 vs day4   1  793.5   793.5   8.949 0.0152 *
  day: day 1+2 vs 3    1  270.8   270.8   3.054 0.1145  
  day: day 1+2+3 vs 4  1                                
Residuals              9  798.0    88.7                 
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

                      Df Sum Sq Mean Sq F value Pr(>F)  
day                    3 1106.9   369.0   3.908 0.0547 .
  day: day 1 vs day4   1  840.2   840.2   8.898 0.0175 *
  day: day 1+2 vs 3    1  136.7   136.7   1.448 0.2632  
  day: day 1+2+3 vs 4  1  130.0   130.0   1.377 0.2744  
Residuals              8  755.3    94.4                 
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

                      Df Sum Sq Mean Sq F value Pr(>F)  
day                    3 1106.9   369.0   3.908 0.0547 .
  day: day 2 vs day4   1  504.2   504.2   5.340 0.0496 *
  day: day 1+2 vs 3    1   94.6    94.6   1.002 0.3462  
  day: day 1+2+3 vs 4  1  508.2   508.2   5.382 0.0489 *
Residuals              8  755.3    94.4                 
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

                      Df Sum Sq Mean Sq F value Pr(>F)  
day                    3 1106.9   369.0   3.908 0.0547 .
  day: day 1 vs day3   1    0.7     0.7   0.007 0.9351  
  day: day 1+2 vs 3    1   50.0    50.0   0.530 0.4875  
  day: day 1+2+3 vs 4  1 1056.2  1056.2  11.187 0.0102 *
Residuals              8  755.3    94.4                 
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

In [17]:
day<-c("Day1","Day2","Day3","Day4");
sol<-rep(c(1:3),each=4);
sol <- factor(sol)
c1 <- c(13,22,18,39)
c2 <- c(16,24,17,NA)
c3 <- c(5,4,1,22)
y <- c(c1,c2,c3)
data<-data.frame(day,sol,weight);
fit = lm(y ~ day+sol, data)
anova(fit)

Df Sum Sq    Mean Sq    F value  Pr(>F)      
day       3  532.89394 177.631313 22.58025 0.0024607397
sol       2  594.50000 297.250000 37.78602 0.0009593215
Residuals 5   39.33333   7.866667       NA           NA

In [9]:
spaces<-rep(c(1:4),4);
spaces <- rev(spaces)
park<-rep(c(1:4),each=4);
duty<-c("2","4","3","1",
         "3","1","2","4",
         "4","3","1","2",
         "1","2","4","3");
obs<-c(29,44,54,71,
       22,22,59,100,
       38,31,40,79,
       29,27,83,100);
Q2<-data.frame(spaces,park,duty,obs);
Q2.mod<-lm(obs~ factor(spaces) + factor(park) + factor(duty),data=Q2);
anova(Q2.mod);

Df Sum Sq Mean Sq    F value   Pr(>F)      
factor(spaces) 3  9025.0 3008.33333 71.062992 4.440579e-05
factor(park)   3   370.5  123.50000  2.917323 1.225065e-01
factor(duty)   3  1389.5  463.16667 10.940945 7.591484e-03
Residuals      6   254.0   42.33333        NA           NA

        df
squre   1
row     6
col     6
error   3
treatment 15
total   31